src/components/LandEnsPage/LandSetNameForm/LandSetNameForm.tsx

```diff
@@ -0,0 +1,154 @@
+      })
+    }
+    if (isWaitingTxSetContent) {
+      this.setState({
+        isSetResolverDone: true, 
+        isSetContentDone: true, 
+        isSetResolverDisabled: true,
+        isSetContentDisabled: true
+      })
+    }
+  }
+
+  handleSetResolver = () => {
+    const { onSetENSResolver, land, selectedName } = this.props
+    onSetENSResolver(selectedName, land)
+    this.setState({
+      isSetResolverDone: true,
+      isSetResolverDisabled: true,
+      isSetContentDisabled: false,
+    })
+  }
+  
+  handleSetContent = () => {
+    const { onSetENSContent, land, selectedName } = this.props
+    onSetENSContent(selectedName, land)
+    this.setState({
+      isSetContentDone: true,
+      isSetResolverDisabled: true,
+      isSetContentDisabled: true,
+    })
+  }
+ 
+  handleBack = () => {
+    this.props.onRestartForm()
+  }
+
+  render() {
+    const { isLoading, isWaitingTxSetContent, isWaitingTxSetResolver, land, error, onNavigate } = this.props
+    const { 
+      isSetContentDisabled,
+      isSetResolverDisabled,
+      isSetContentDone,
+      isSetResolverDone,
+    } = this.state
+
+    const isSubmitDisabled = !(isSetContentDisabled && isSetResolverDisabled) || isLoading || isWaitingTxSetResolver || isWaitingTxSetContent
+    const isBackDisabled = isSetResolverDone || isSetContentDone
+    const isResolverLoading = (isLoading && isSetResolverDone && !isSetContentDone) || isWaitingTxSetResolver 
+    const isContentLoading = (isLoading && isSetResolverDone && isSetContentDone) || isWaitingTxSetContent 
+ 
+    let buttonSetResolver = <> 
+      {isSetResolverDone? t('global.approved_tx'): t('global.send_tx')} 
+      {!(isResolverLoading && (isLoading || isWaitingTxSetResolver)) && isSetResolverDone && <Icon name='check' />} 
+    </>
+    let buttonSetContent = <> 
+      {isSetContentDone? t('global.approved_tx'): t('global.send_tx')} 
+      {!isContentLoading && isSetContentDone && <Icon name='check' />} 
+    </>
+    let setResolverButtonClassName = ((isSetResolverDone && !(isLoading || isWaitingTxSetResolver)) || isSetContentDone) ? 'grey-button': ''
+    let setContentButtonClassName = (isSetContentDone && !(isLoading || isWaitingTxSetContent)) ? 'grey-button': ''
+    let isSetResolverButtonDisabled = isSetResolverDisabled || isLoading || isWaitingTxSetResolver || isWaitingTxSetContent
+    let isSetContentButtonDisabled = isSetContentDisabled || isLoading || isWaitingTxSetResolver || isWaitingTxSetContent
+    if (error && error.code === 4001) {
+      if (error.origin === 'SET_ENS_RESOLVER') {
+        buttonSetResolver = <> { t('global.retry_tx') } </>
+        isSetResolverButtonDisabled = false
+        isSetContentButtonDisabled = true
+        setResolverButtonClassName = ''
+      } else if (error.origin === 'SET_ENS_CONTENT') {
+        buttonSetContent =  <> { t('global.retry_tx') } </>
+        isSetContentButtonDisabled = false
+        setContentButtonClassName = ''
+      }
+    }
```

Comment:
I think this code could be extracted a bit. Let me know if you think these suggestions make sense:
1. We could name a few of these conjoined conditions and then reuse them. For example: 
```ts
// Test code, do not copy :P
isLoading() {
  return isLoading || isWaitingTxSetResolver || isWaitingTxSetContent
}
(...)
const isSubmitDisabled = !(isSetContentDisabled && isSetResolverDisabled) || this.isLoading()
let isSetResolverButtonDisabled = isSetResolverDisabled || this.isLoading()
let isSetContentButtonDisabled = isSetContentDisabled  || this.isLoading()
```

2. Commented below as well but it's a good idea to just use the actual action here instead of the string that represents it, so if the action changes, this code will still work and wont fail silently (`'SET_ENS_RESOLVER' => SET_ENS_RESOLVER_FAILURE` for example)

3. I'm a bit confused by the use of the `state` here. Shouldn't it all come from redux? why are you setting it here also? Correct me if I'm wrong but it seems like you have all the information you need on the state and you just have to feed it to the container?

4. **This is just a suggestion i have no idea if it's a good idea**: If you're having two "boxes" (Resolver and Content) that have similar behaviours but different sources for data, maybe you're looking at a missing abstraction. You could have a `TransactionBox` Component (maybe a bad name) which is agnostic to the situation but does all the heavy lifting, and you can then feed the booleans from here.